internal/metamorph/store/postgresql/postgres.go

```diff
@@ -191,6 +191,38 @@ func (p *PostgreSQL) Get(ctx context.Context, hash []byte) (*store.StoreData, er
 	return data, nil
 }
 
+// GetRawTxs implements the MetamorphStore interface. It attempts to get rawTxs for given hashes.
+// If the hashes do not exist an empty array is returned, otherwise the retrieved values.
+// If an error happens during the process of getting the results, the error is returned
+// along with already found rawTxs up to the error point.
+func (p *PostgreSQL) GetRawTxs(ctx context.Context, hashes [][]byte) ([][]byte, error) {
+	retRawTxs := make([][]byte, 0)
+
+	q := `SELECT raw_tx
+		FROM metamorph.transactions
+		WHERE hash in (SELECT UNNEST($1::BYTEA[]))`
+
+	rows, err := p.db.QueryContext(ctx, q, pq.Array(hashes))
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
```suggestion
defer rows.Close()
```

The `sql.Rows` returned by `sql.QueryContext()` should always be disposed after usage. I've noticed a few other instances in the file where this isn't done – could you please address those as well?

Since I am not an expert in Golang, I'm uncertain about when they should be closed in case of an error in a transaction. So please follow the same pattern as in the `GetSeenOnNetwork()` function (just add `defer rows.Close()` after `Rollback()`)."

Places to fix:
1. line 209
2. line 564
3. line 649
4. line 779

Reply:
You're right, added, thanks for noticing.